test/packages/parallel/sql_input/agent/input/input.yml.hbs

```diff
@@ -1,4 +1,6 @@
 metricsets: ["query"]
+data_stream:
+  dataset: {{data_stream.dataset}}
```

Comment:
This seems to be required in input packages to actually leverage this option. I think we should handle this in a different way. Ideally Fleet should add this if this is required in all input packages. cc @kpollich @mrodm

Reply:
Yes I think you are right. Fleet is enforcing this as a requirement, so Fleet should add this to the input package if it isn't explicitly set as a "default". Dynamically altering the handlebars template before rendering it isn't something we've done before in Fleet, but I don't think it will be more than some string manipulation. There's potential for complexity, though, to be sure. I created https://github.com/elastic/kibana/issues/184008 to capture this on the Fleet side.

Reply:
Thanks :+1:

I think this config could be appended if not already set.